diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter6/InlineClassesWarnTest.kt

```diff
@@ -0,0 +1,126 @@
+import org.junit.jupiter.api.Test
+
+class InlineClassesWarnTest : LintTestBase(::InlineClassesRule) {
+    private val ruleId = "$DIKTAT_RULE_SET_ID:inline-classes"
+
+    @Test
+    @Tag(WarningNames.INLINE_CLASS_CAN_BE_USED)
+    fun `should not trigger on inline class`() {
+        lintMethod(
+            """
+                |inline class Name(val s: String) {}
+            """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.INLINE_CLASS_CAN_BE_USED)
+    fun `should trigger on regular class`() {
+        lintMethod(
+            """
+                |class Some {
+                |   val config = Config()
+                |}
+            """.trimMargin(),
+            LintError(1, 1, ruleId, "${INLINE_CLASS_CAN_BE_USED.warnText()} class Some", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.INLINE_CLASS_CAN_BE_USED)
+    fun `should trigger on class with appropriate modifiers`() {
+        lintMethod(
+            """
+                |final class Some {
+                |   val config = Config()
+                |}
+            """.trimMargin(),
+            LintError(1, 1, ruleId, "${INLINE_CLASS_CAN_BE_USED.warnText()} class Some", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.INLINE_CLASS_CAN_BE_USED)
+    fun `should not trigger on class with inappropriate modifiers`() {
+        lintMethod(
+            """
+                |abstract class Some {
+                |   val config = Config()
+                |}
+            """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.INLINE_CLASS_CAN_BE_USED)
+    fun `should trigger on class with val prop in constructor`() {
+        lintMethod(
+            """
+                |class Some(val anything: Int) {
+                |
+                |}
+            """.trimMargin(),
+            LintError(1, 1, ruleId, "${INLINE_CLASS_CAN_BE_USED.warnText()} class Some", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.INLINE_CLASS_CAN_BE_USED)
+    fun `should not trigger on class with var prop #1`() {
+        lintMethod(
+            """
+                |class Some(var anything: Int) {
+                |
+                |}
+            """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.INLINE_CLASS_CAN_BE_USED)
+    fun `should not trigger on class with var prop #2`() {
+        lintMethod(
+            """
+                |class Some {
+                |   var some = 3
+                |}
+            """.trimMargin()
+        )
+    }
+
+    @Disabled
+    @Test
+    @Tag(WarningNames.INLINE_CLASS_CAN_BE_USED)
+    fun `should not trigger on class that extends class or interface`() {
+        lintMethod(
+            """
+                |class Some : Any {
+                |   val some = 3
+                |}
+            """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.INLINE_CLASS_CAN_BE_USED)
+    fun `should not trigger on class with internal constructor`() {
```

Comment:
Are you sure it's prohibited? IDEA doesn't seem to highlight it in red.

Reply:
idk, my idea doesn't :)
![image](https://user-images.githubusercontent.com/23018090/105183808-a9840180-5b3f-11eb-9ca4-e5fc50005d02.png)

Reply:
Strange, I have this 
![image](https://user-images.githubusercontent.com/48321920/105187609-05e92000-5b44-11eb-86c8-260465e23179.png)
IntelliJ IDEA 2020.3 (Ultimate Edition)
Build #IU-203.5981.155, built on December 1, 2020

Reply:
hmm, do you try this code in diktat? I would expect results to be the same with same kotlin version. Which version of kotlin plugin do you have in idea? My is 203-1.4.30-RC-IJ6682.9 for idea 2020.3.2. Let's check if this code compiles with kotlin 1.4.21.

Reply:
Now I have 1.4.21 and this code doesn't compile

Reply:
OK, that is a bug in IDEA plugin. Maybe, it'll be allowed in 1.4.30, but let's allow only public for now.